trim.production.js

```diff
@@ -13,6 +13,39 @@ var token;
  * @param {String} recordNo The TRIM RecordNo of the new document, if created.
  */
 
+/**
+ *
+ * @param {Object} body
+ * @param {String} body.Title
+ * @param {String} body.Container
+ * @param {String} body.RecordExtension
+ * @param {String[]} body.AlternativeContainers
+ * @param {String} body.Record
+ * @return {String|null} Error message. null if no errors are found
+ */
+function getCreateRecordBodyErrors(body) {
+
+  if (!body.Title) return 'A title is required'
+
+  if (!body.Container) return 'A container is required'
+
+  if (!body.RecordExtension) return 'An extension is required'
+  if (!R.test(/^[a-zA-Z]+$/, body.RecordExtension)) return 'Record extensions must only contain letters'
+
+  // a dataUrl header ends with a comma
+  const isProbablyDataUrl = R.pipe(
+      R.slice(0, 200),  // prevent checking the *entire* string
+      R.contains(',')
+  )
+  const validBase64Characters = /^[A-Za-z0-9+/=]+$/
```

Comment:
I was going to use this one http://stackoverflow.com/a/475217/1787372 but I decided that I don't know if that is 100% accurate 100% of the time, and don't want to try to validate that. I figured that just validating that there are no non-base64 characters is _good enough_ for a real basic validity check.